rfc/src/rfcs/0007-global-conditions.md

```diff
@@ -0,0 +1,111 @@
+  - `<reason>` is an explanation that depends on the status of the global condition.
+
+For example, let's assume we implement the option requested in [#2299](https://github.com/model-checking/kani/issues/2299).
+A concrete example of this output would be:
+
+```
+GLOBAL CONDITIONS:
+ - `fail_uncoverable`: SUCCESS (all cover statements were satisfied as expected)
+```
+
+A `FAILED` status in any enabled global condition will cause verification to fail.
+In that case, the overall verification result will point out that one or more global conditions failed, as in:
+
+```
+VERIFICATION:- FAILURE (one or more global conditions failed)
+```
+
+This last UI change will also be implemented for the terse output.
+Finally, checks that cause an enabled global condition to fail will be reported using the same interface we use for failed checks[^failed-checks].
+
+**Global conditions which aren't enabled won't appear in the `GLOBAL CONDITIONS` section**.
+Their status will be computed regardless[^status-computation], and we may consider showing this status when the `--verbose` option is passed.
+
+The documentation of global conditions will depend on how they're enabled, which depends on a case-by-case basis.
+However, we may consider adding a new subsection `Global conditions` to the `Reference` section that collects all of them so it's easier for users to consult all of them in one place.
+
+## Detailed Design
+
+The only component to be modified is `kani-driver` since that's where verification results are built and determined.
+But we should consider moving this logic into another crate.
+
+We don't need new dependencies.
+The corner cases will depend on the specific global conditions to be implemented.
+
+## Rationale and alternatives
+
+As mentioned earlier, we're proposing this change to help users understand global conditions and how they're determined.
+In many cases, global conditions empower users to write harnesses which weren't possible to write before.
+As an example, the `#[kani::should_panic]` attribute allowed users to write harnesses expecting panic-related failures.
+
+Also, we don't really know if more global conditions will be requested in the future.
+We may consider discarding this proposal and waiting for the next feature that can be implemented as a global condition to be requested.
+
+### Alternative: Global conditions as regular checks
+
+One option we've considered in the past is to enable global conditions as a regular checks.
+While it's technically doable, it doesn't feel appropriate for global conditions to reported through regular checks since generally a higher degree of visibility may be appreciated.
+
+## Open questions
+
+No open questions.
+
+## Future possibilities
+
+A redesign of Kani's output is likely to change the style/architecture to report global conditions.
+
+[^status-computation]: The results for global conditions would be computed during postprocessing based on the results of other checks.
```

Comment:
so the status of a global condition could differ when the switch `--fail-uncoverable` is activated vs not activated ?

Reply:
No. The status computed for the condition itself won't differ. But activating the switch will simply cause the status of the global condition to affect the overall verification result.